lib/internal/crypto/random.js

```diff
@@ -291,20 +292,42 @@ function getRandomValues(data) {
 // Implements an RFC 4122 version 4 random UUID.
 // To improve performance, random data is generated in batches
 // large enough to cover kBatchSize UUID's at a time. The uuidData
-// and uuid buffers are reused. Each call to randomUUID() consumes
-// 16 bytes from the buffer.
-
-const kHexDigits = [
-  48, 49, 50, 51, 52, 53, 54, 55,
-  56, 57, 97, 98, 99, 100, 101, 102
-];
+// buffer is reused. Each call to randomUUID() consumes 16 bytes
+// from the buffer.
 
 const kBatchSize = 128;
 let uuidData;
 let uuidNotBuffered;
-let uuid;
 let uuidBatch = 0;
 
+let hexBytesCache;
+function getHexBytes() {
+  if (hexBytesCache === undefined) {
+    hexBytesCache = new Array(256);
+    for (let i = 0; i < hexBytesCache.length; i++) {
+      hexBytesCache[i] = i.toString(16).padStart(2, '0');
```

Comment:
```suggestion
      hexBytesCache[i] = NumberPrototypePadStart(NumberPrototypeToString(i, 16), 2, '0');
```

Reply:
Thanks! Should primordials be always used directly, or is there some exception?

Reply:
We should use them whenever we can ;)